osu.Game.Rulesets.Osu/OsuInputManager.cs

```diff
@@ -19,20 +19,20 @@ public bool AllowUserPresses
             set => ((OsuKeyBindingContainer)KeyBindingContainer).AllowUserPresses = value;
         }
 
-        protected override RulesetKeyBindingContainer CreateKeyBindingContainer(RulesetInfo ruleset, int variant, SimultaneousBindingMode unique)
-            => new OsuKeyBindingContainer(ruleset, variant, unique);
+        protected override RulesetKeyBindingContainer CreateKeyBindingContainer(RulesetInfo ruleset, int variant, SimultaneousBindingMode all)
+            => new OsuKeyBindingContainer(ruleset, variant, all);
```

Comment:
in what way is changing the binding mode from unique to all game breaking. It seems to work just fine. Does it somehow make abuse much easier. If this can't be the case then I think the osu actions should have four actions for each of the possible keys, because even ignoring my mod, I mapped the two button2s to my primary keys, and streams just didn't, but I didn't know why (I thought I was just doing bad). I imagine this could easily happen to other if the fact that the keys are exclusive is not apparent to the users.

Reply:
Yes, the limitation you state here is *intended*. One game binding is only supposed to actuate once. This is how osu! works and *why* the `Unique` mode is specified here.

I would highly suggest understanding the underlying concepts before making changes like this.

For a mod like this, may I recommend using the taiko bindings the user has set instead? This will require some larger ruleset-level changes to allow the mod to expose to the ruleset which key it intends on using during gameplay, though.

Reply:
I still don't quite understand why it matters, but I guess if I look more into the key binding stuff or the framework, I may be able to find some reason it could cause problems. I'll look into making that change. Though I think this limitation somehow should be made clear to the user because as it stands its not clear to the user that this is how it works (as I was having problems with streams and burst for awhile because I was using two button2s and I just assumed I sucked or that lazer somehow had different hit detection or something. I really appreciate you taking time to give me feedback here, as I understand you are very busy (great job by the way). I'll look into making some of these changes, but it might be a while (IDK) because I'm starting up school soon. You should probably mark this WIP.